kubernetes/argo.libsonnet

```diff
@@ -1,5 +1,44 @@
 local ok = import 'outreach.libsonnet';
-{
+
+// namespace for argocd
+local argocdNamespace = 'argocd';
+
+{ 
+  Application(name): ok._Object('argoproj.io/v1alpha1', 'Application', name, namespace=argocdNamespace) {
+    local this = self,
+
+    namespace_:: error 'namespace_ is required',
+    path_:: error 'path_ is required',
+    repo_:: error 'repo_ is required',
+    initial_revision_:: 'none',
+    repo_name_:: std.split(this.repo_, "/")[std.length(std.split(this.repo_, "/"))-1],
+    source_path_:: std.join("/", std.slice(std.split(this.path_, "/"), 0, std.length(std.split(this.path_, "/"))-1, 1)),
+    env_:: {},
```

Comment:
nit: these are all "first class arguments" so we can drop the `_` which implies it's overriding something else.

At the same time, there's also an argument to be made that these could/should be arguments provided to the function. There's less of a standard about that though, so I'm OK not going that route if we want to.

Reply:
I think it looks cleaner with out the `_` anyways so i agree we should remove it 👍